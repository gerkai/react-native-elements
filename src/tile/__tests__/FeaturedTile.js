import React from 'react';
import {shallow} from 'enzyme';
import FeaturedTile from '../FeaturedTile';

describe('FeaturedTitle component', () => {
  it('should render without issues', () => {
    const component = shallow(<FeaturedTile imageSrc={{url: 'http://google.com'}}/>);

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });
});
