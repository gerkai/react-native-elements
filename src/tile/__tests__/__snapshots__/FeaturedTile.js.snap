// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedTitle component should apply values from theme 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 600,
      "opacity": 1,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "height": 600,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "url": "http://google.com",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 600,
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "rgba(0,0,0,0.2)",
          "bottom": 0,
          "flex": 1,
          "justifyContent": "center",
          "left": 0,
          "paddingBottom": 40,
          "paddingLeft": 25,
          "paddingRight": 25,
          "paddingTop": 45,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "justifyContent": "center",
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
            "color": "#ffffff",
            "fontSize": 27.5,
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
        testID="featuredTileTitle"
        theme={
          Object {
            "FeaturedTile": Object {
              "title": "I am featured",
            },
            "colors": Object {
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
            },
          }
        }
        updateTheme={[Function]}
      >
        I am featured
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
            "color": "#ffffff",
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
        theme={
          Object {
            "FeaturedTile": Object {
              "title": "I am featured",
            },
            "colors": Object {
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
            },
          }
        }
        updateTheme={[Function]}
      />
    </View>
  </View>
</View>
`;

exports[`FeaturedTitle component should render with Icon 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "height": 70,
      "width": 750,
    }
  }
>
  <ImageBackground
    resizeMode="cover"
    source={
      Object {
        "url": "http://google.com",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "height": 70,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "rgba(0,0,0,0.2)",
          "bottom": 0,
          "flex": 1,
          "height": 70,
          "justifyContent": "center",
          "left": 0,
          "paddingBottom": 40,
          "paddingLeft": 25,
          "paddingRight": 25,
          "paddingTop": 45,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "height": 70,
            "justifyContent": "center",
          }
        }
      >
        <Themed.Icon
          name="play-circle"
          type="font-awesome"
        />
      </Component>
      <Themed.Text
        h4={true}
        style={
          Object {
            "backgroundColor": "yellow",
            "color": "#ffffff",
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
        testID="featuredTileTitle"
      />
      <Themed.Text
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
            "color": "blue",
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
      />
    </Component>
  </ImageBackground>
</TouchableOpacity>
`;

exports[`FeaturedTitle component should render with width and height 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  height={20}
  style={
    Object {
      "height": 20,
      "width": 34,
    }
  }
  width={34}
>
  <ImageBackground
    resizeMode="cover"
    source={
      Object {
        "url": "http://google.com",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "height": 20,
        "justifyContent": "center",
        "width": 34,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "rgba(0,0,0,0.2)",
          "bottom": 0,
          "flex": 1,
          "justifyContent": "center",
          "left": 0,
          "paddingBottom": 40,
          "paddingLeft": 25,
          "paddingRight": 25,
          "paddingTop": 45,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "justifyContent": "center",
          }
        }
      />
      <Themed.Text
        h4={true}
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
            "color": "#ffffff",
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
        testID="featuredTileTitle"
      />
      <Themed.Text
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
            "color": "#ffffff",
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
      />
    </Component>
  </ImageBackground>
</TouchableOpacity>
`;

exports[`FeaturedTitle component should render without issues 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "height": 600,
      "width": 750,
    }
  }
>
  <ImageBackground
    resizeMode="cover"
    source={
      Object {
        "url": "http://google.com",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "height": 600,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "rgba(0,0,0,0.2)",
          "bottom": 0,
          "flex": 1,
          "justifyContent": "center",
          "left": 0,
          "paddingBottom": 40,
          "paddingLeft": 25,
          "paddingRight": 25,
          "paddingTop": 45,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "justifyContent": "center",
          }
        }
      />
      <Themed.Text
        h4={true}
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
            "color": "#ffffff",
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
        testID="featuredTileTitle"
      />
      <Themed.Text
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
            "color": "#ffffff",
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
      />
    </Component>
  </ImageBackground>
</TouchableOpacity>
`;
